src/main.rs

```diff
@@ -1,15 +1,17 @@
-use beacon_api_client::{Client, StateId};
+pub mod api_client;
 use url::Url;
+pub mod cli;
+pub mod error;
+pub mod serde;
+pub mod types;
+use clap::Parser;
 
 #[tokio::main]
 async fn main() {
-    let s = "http://127.0.0.1:8003/";
-    let url: Url = Url::parse(s).unwrap();
-    let client = Client::new(url);
-
-    let checkpoints = client.get_finality_checkpoints(StateId::Finalized).await.unwrap();
-
-    println!("previous checkpoint: {:?}", checkpoints.previous_justified);
-    println!("current checkpoint: {:?}", checkpoints.current_justified);
-    println!("finalized checkpoint: {:?}", checkpoints.finalized);
+    // read in args from CLI
+    let args = cli::CliConfig::parse();
+    // instantiate client and pass to run_cli
+    let url: Url = Url::parse(&args.endpoint).unwrap();
+    let client = api_client::Client::new(url);
+    cli::run_cli(client, args).await;
```

Comment:
if we make the above adjustments:

(also stylistically, I'd avoid comments that are just English representations of the code)

and you don't need to type hint `url` as the rust compiler can infer that from `Url::...` in that line

```suggestion
    let args = CliConfig::parse();
    let url = Url::parse(&args.endpoint).unwrap();
    let client = Client::new(url);
    run_cli(&client, &args).await;
```